### PR TITLE
fixes #4967 feat(reporting): Add ReportLog model

### LINKED ISSUE #4967: [Reporting] Create ReportLog Model within Experimenter
ReportLog model will need to include fields listed in the blueprint

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-1159)
┆Issue Number: EXP-1159

### PR DESCRIPTION
**Pull request by tiftran**
Because:

* Model will be used to generate reports

This Commit:

* Creates the ReportLog Model, tests and migrations
* ReportLog Factory for testing

### CODE REVIEW

File: app/experimenter/reporting/constants.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    EXPERIMENT_STATUS_CHOICES = tuple(
+        sorted(
+            list(
+                set(
+                    tuple(NimbusConstants.Status.choices)
+                    + ExperimentConstants.STATUS_CHOICES
+                )
+            )
+        )
+    )
```
Comment (jaredlockhart) [inline app/experimenter/reporting/constants.py:17]: Yeah you could do it this way, but also with constants sometimes it's also fine to just explicitly refer them individually so it's a bit clearer 🤷‍♀️

File: app/experimenter/reporting/constants.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    EXPERIMENTS_TYPE = (
+        ("Normandy", "Normandy"),
+        ("Nimbus-Desktop", "Nimbus-Desktop"),
+        (
+            "Nimbus-Mobile",
+            "Nimbus-Mobile",
```
Comment (jaredlockhart) [inline app/experimenter/reporting/constants.py:23]: We recently added iOS in addition to Fenix(Android) so I'm not sure whether you want to lump both iOS/Android together here or keep them separate?

File: app/experimenter/reporting/constants.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        STATUS_CHANGE = "Status Change"
+        PAUSE = "Pause"
+        RECIPE_CHANGE = "Recipe Change"
+        QA_LAUNCH = "QA Launch"
+        LAUNCH = "Launch"
+        EXPERIMENT_COMPLETE = "Experiment Complete"
+        FILTERING_ISSUE = "Filtering Issue"
+        ENROLLMENT_ISSUE = "Enrollment Issue"
+        EXPERIMENT_DESIGN_ISSUE = "Experiment Design Issue"
+        RESULTS_ADDED = "Results Added"
```
Comment (jaredlockhart) [inline app/experimenter/reporting/constants.py:45]: Neat!

**Review by jaredlockhart — APPROVED** [review]
Neat!  This is really interesting @tiftran I can't wait to see these awesome reports with breakdowns of experiment timeslines and outcomes :D

**Comment by jaredlockhart** [comment]
OH also @tiftran heads up we took out the `must be up to date with main` requirement so now you have to be really careful about PRs with migrations so make sure you rebase on main and repush before you merge 👍

**Comment by jaredlockhart** [comment]
Also @tiftran don't forget to add the `Data Models` tag to this.

File: app/experimenter/reporting/constants.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        ),
+    )
+
+    class Event(models.TextChoices):
+        CREATE = "Create"
+        UPDATE = "Update"
+        END = "End"
+
+    class EventReason(models.TextChoices):
+        NEW = "New"
```
Comment (jaredlockhart) [inline app/experimenter/reporting/constants.py:33]: LEGACY = (None, Legacy.Status.DRAFT)
NIMBUS = (None, Nimbus.Status.DRAFT, Nimbus.PublishStatus.IDLE)


clone_change = experiment.changes.filter(**Changes.CLONE_LEGACY)

**Comment by jaredlockhart** [comment]
@tiftran you gotta rebase this on main to get the new circle definitions

**Comment by tiftran** [comment]
@jaredlockhart will do, currently working on the suggestions you've made 🙂

**Pull request merged**